File: frontend/src/api/board.ts
import { apiGet } from '@/lib/api-client';
import type { ApplicationFilters } from '@/types/application';
import type { SourceCoverage } from '@/types/search';

/** The board filters the summary honors, same names and semantics as the
    /applications list, so the rail counts and the list share predicates. */
export type BoardSummaryFilters = Pick<
  ApplicationFilters,
  | 'search'
  | 'location'
  | 'location_strict'
  | 'salary_min'
  | 'salary_max'
  | 'salary_currency'
  | 'is_remote'
  | 'first_quest_ok'
  | 'posted_within_days'
  | 'found_within_days'
  | 'timezone_name'
>;

export interface KindSummary {
  id: string;
  label: string;
  sub: string;
  hue: string;
  order: number;
  count: number;
  new_today: number;
}

export interface BoardSummary {
  total: number;
  new_today: number;
  /** last healthy quest-source run (ISO, naive UTC); null before the first */
  checked_at: string | null;
  /** last fully completed Find Work pull; partial source responses do not count */
  career_checked_at: string | null;
  /** last healthy Side Quest source run */
  side_quest_checked_at: string | null;
  career_refresh: CareerRefreshReceipt | null;
  kinds: KindSummary[];
}

export interface CareerRefreshReceipt {
  run_id: string;
  status: 'pending' | 'running' | 'completed' | 'failed';
  started_at: string | null;
  completed_at: string | null;
  jobs_found: number;
  new_jobs: number;
  error: string | null;
  source_coverage: SourceCoverage | null;
}

export function getBoardSummary(filters: BoardSummaryFilters = {}): Promise<BoardSummary> {
  return apiGet<BoardSummary>('/board/summary', { ...filters });
}

File: backend/app/schemas/board.py
"""Board summary shapes: per-kind supply counts the board rail renders."""

from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel

from app.schemas.search import SourceCoverage


class KindSummary(BaseModel):
    id: str
    label: str
    sub: str
    hue: str
    order: int
    count: int
    new_today: int


class CareerRefreshReceipt(BaseModel):
    run_id: str
    status: str
    started_at: datetime | None = None
    completed_at: datetime | None = None
    jobs_found: int = 0
    new_jobs: int = 0
    error: str | None = None
    source_coverage: SourceCoverage | None = None


class BoardSummaryResponse(BaseModel):
    total: int
    new_today: int
    # last time a quest source ran healthy, from the scrape run log; None
    # until the first refresh
    checked_at: datetime | None = None
    # Career freshness is a COMPLETED whole pull, not the newest individual
    # source response. Side quests retain their independent source cadence.
    career_checked_at: datetime | None = None
    side_quest_checked_at: datetime | None = None
    # Newest Find Work attempt, including the exact per-source completion
    # receipt when available. This survives page reloads and app restarts.
    career_refresh: CareerRefreshReceipt | None = None
    kinds: list[KindSummary]
